src/main/java/com/powsybl/openloadflow/sensi/AbstractSensitivityAnalysis.java

```diff
@@ -66,11 +172,9 @@ protected JacobianMatrix createJacobianMatrix(EquationSystem equationSystem, Vol
         // Wrap factors in specific class to have instant access to their branch and their equation term
         private final Object context;
 
-        private final String functionId;
-
         private final String variableId;
 
-        protected final LfBranch functionLfBranch;
+        ILfSensitivityFunction function;
```

Comment:
LfBranch and LfBus has a common base class LfElement. Would it be possible to use it here in order to avoid adding ILfSensitivityFunction?

Reply:
Maybe we could try in this PR to adapt LfFactor to internal API. To avoid LfFactor subclassing hell (with all function/variable combinations), we could try to only have one LfFactor class and one LfFactorGroup. So having LfElement type for variable and function.

Reply:
So, we would to store a LfElement + a SensitivityFunctionType instead of having a class wrapping the element for each Function type ? And then, to get the equation term from that, we write a big switch statement, instead of overriding methods ? (and do the same thing for the variables)